PowerShell/ScubaGear/Testing/Unit/PowerShell/Orchestrator/Invoke-ScubaConfig.Tests.ps1

```diff
@@ -0,0 +1,142 @@
+            Mock -ModuleName Orchestrator Invoke-ReportCreation {}
+            Mock -ModuleName Orchestrator Disconnect-SCuBATenant {}
+            Mock -CommandName New-Item {}
+            Mock -CommandName Copy-Item {}
+        }
+        Context 'Testing Invoke-Scuba with -ConfigFilePath arg and parameter override' {
+            BeforeAll {
+            
+                [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'SplatParamsRef')]
+
+                # Set up the reference comparison withunmodified parameters by reading the hash from the config file
+                $ConfigFile = ( Join-Path -Path $PSScriptRoot  -ChildPath "orchestrator_config_test.yaml" )
+                $Content = Get-Content -Raw -Path $ConfigFile
+                $ScubaConfRef = $Content | ConvertFrom-Yaml
+                # Set up the splat params to point to this file ( this will be cloned and modified for overrides )
+                $SplatParamsRef = @{
+                    ConfigFilePath = $ConfigFile
+                }
+
+                # Set up the splat params to refernce a config file without authentication parameters
+                # but supply then expicitly as arguments.  Use the authentication values from the refrence above
+
+                $ConfigFileNoCreds = ( Join-Path -Path $PSScriptRoot  -ChildPath "orchestrator_config_test_no_creds.yaml" )
+                $SplatParamsCreds = @{
+                    ConfigFilePath = $ConfigFileNoCreds
+                    Organization = $ScubaConfRef.Organization
+                    AppID = $ScubaConfRef.AppID
+                    CertificateThumbprint = $ScubaConfRef.CertificateThumbprint
+                }
+
+                # General function to to compare overrides
+                function OverrideTest( $ModKey, $ModValue) {
+                    # Add the modified parameter as passed in parameter
+                    $SplatParams = $SplatParamsRef.Clone()
+                    $SplatParams[$ModKey] = $ModValue
+                    Invoke-Scuba @SplatParams
+                    # The values setup by the override are propegated by the orchestrator
+                    # into the Scuba config singleton
+                    $ConfTest = [ScubaConfig]::GetInstance().Configuration
+                    $pass = $true
+                    if ( (Compare-Object @($ScubaConfRef.keys) @($ConfTest.keys)))
+                    {
+                        $pass = $false
+                    }
+                    else
+                    {
+                        # Test all values to be sure that the override parameter matches modified value
+                        # The other values should be the same
+                        foreach ($key in $ConfTest.keys )
+                        {
+                            if ( $key -eq $Modkey ) {
+                                $isDifferentValue = (( Compare-Object  $ModValue $ConfTest.$key ) -ne $none )
```

Comment:
What is '$none'?  It looks to me as if it is never set to anything. Did you mean the automatic vaariable '$null'?